include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -669,6 +669,18 @@ get(const oneapi::dpl::__internal::tuple<_Tp...>&& __a)
 {
     return ::std::move(__a).template get<_Idx>();
 }
+
+// To enable oneapi::dpl::zip_iterator to satisfy the requirements for the std::input_iterator concept in C++20,
+// we must provide a basic_common_reference specialization for our internal tuple implementation.
+#if _ONEDPL_CPP20_CONCEPTS_PRESENT
```

Comment:
It's not obvious to me why the presence of ``std::basic_common_reference`` depends on concepts. Is not it enough to check for C++20?

Reply:
The reasoning here is that `std::basic_common_reference` is introduced in [P0898R3](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0898r3.pdf) which corresponds to `__cpp_lib_concepts`.

If we just check the C++ version and not the `__cpp_lib_concepts` feature macro, then we may be use a compiler that has C++20 support but the standard library version may not support library concepts (ex: libstdc++ 9).

Reply:
I see, thanks. Ok to keep it as you did, but a short comment would be useful.

Reply:
A comment has been added clarifying why we guard the specialization with the concepts support macro. Also, I have kept the `__cpp_concepts` check in the macro since trying to use the concepts library without a compiler that supports concepts is problematic.